sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -524,6 +524,16 @@ public static Read<PubsubMessage> readMessagesWithAttributesAndMessageId() {
         .build();
   }
 
+  /**
```

Comment:
I believe to support bounded writes, on line 1329 of this file you'll need to add explicit copying of the orderingKey as well (in the `output.add(OutgoingMessage.of(...` section of `processElement`